pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -0,0 +1,157 @@
+           - Its dimensions are :math:`d_{n-1,0}` and :math:`d_{n-1,1}`, representing the auxiliary dimension from the previous site and the physical dimension of the site.
+
+        3. **Intermediate tensors**:
+
+           - All tensors in the middle (i.e., :math:`A^{(j)}` for :math:`2 \leq j \leq n-1`) have **rank 3** (three-dimensional tensors).
+           - Their dimensions are :math:`d_{j,0}, d_{j,1},` and :math:`d_{j,2}`, where:
+
+             - :math:`d_{j,0}`: Bond dimension connecting to the previous tensor.
+             - :math:`d_{j,1}`: Physical dimension for the site.
+             - :math:`d_{j,2}`: Bond dimension connecting to the next tensor.
+
+           - Importantly, the bond dimensions match between adjacent tensors:
+
+             .. math::
+
+                d_{j-1,2} = d_{j,0}.
+
+        Additionally, the physical dimension of the site should always be fixed at :math:`2`,
+        while the other dimensions must be powers of two.
+        The following input is valid:
+
+        .. code-block::
+
+            mps = [
+                np.array([[0.0, 0.107], [0.994, 0.0]]),
+                np.array(
+                    [
+                        [[0.0, 0.0, 0.0, -0.0], [1.0, 0.0, 0.0, -0.0]],
+                        [[0.0, 1.0, 0.0, -0.0], [0.0, 0.0, 0.0, -0.0]],
+                    ]
+                ),
+                np.array(
+                    [
+                        [[-1.0, 0.0], [0.0, 0.0]],
+                        [[0.0, 0.0], [0.0, 1.0]],
+                        [[0.0, -1.0], [0.0, 0.0]],
+                        [[0.0, 0.0], [1.0, 0.0]],
+                    ]
+                ),
+                np.array([[-1.0, -0.0], [-0.0, -1.0]]),
+            ]
+
+        The dimensions of ``mps`` are: :math:`[(2,2), (2,2,4), (4,2,2), (2,2)]`, that satisfy the criteria described above.
+    """
+
+    def __init__(self, mps, wires, id=None):
+        super().__init__(*mps, wires=wires, id=id)
+
+    @property
+    def mps(self):
+        """list representing the MPS input"""
+        return self.data
+
+    def _flatten(self):
+        hyperparameters = (("wires", self.wires),)
+        return self.mps, hyperparameters
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        hyperparams_dict = dict(metadata)
+        return cls(data, **hyperparams_dict)
+
+    def map_wires(self, wire_map):
+        new_wires = Wires([wire_map.get(wire, wire) for wire in self.wires])
+        return MPSPrep(self.mps, new_wires)
+
+    @classmethod
+    def _primitive_bind_call(cls, mps, wires, id=None):
+        # pylint: disable=arguments-differ
+        if cls._primitive is None:
+            # guard against this being called when primitive is not defined.
+            return type.__call__(cls, mps=mps, wires=wires, id=id)  # pragma: no cover
+        return cls._primitive.bind(*mps, wires=wires, id=id)
+
+
+if MPSPrep._primitive is not None:  # pylint: disable=protected-access
```

Comment:
Is this indent correct?

Reply:
Yes, it is strange, but that's the way to do it as in [BasisRotation](https://github.com/PennyLaneAI/pennylane/blob/da480d984de2cba094b3295d31398fdd010dc9c5/pennylane/templates/subroutines/basis_rotation.py#L198)